File: 2025-03-04 ABSTRACT E INTERFACE/Dinamica/Misc.ts
export class Misc {

    // Função que recebe uma lista e um valor: retorna um array (caso qtdeRecebida maior que 1) de elementos aleatórios não-repetidos a partir da lista recebida.
    // Foi usado pra setar habilidades aleatórias nos Caçadores e Monstros.
    static retornarElementosAleatorios(listaRecebida:any[], qtdeRecebida:number):any {
        const tamanhoMaximo:number = listaRecebida.length

        if (qtdeRecebida == 1) {
            const indexAleatorio:number = Math.floor(Math.random() * tamanhoMaximo)
            return listaRecebida[indexAleatorio]
        }

        const listaTemporaria:any[] = []
        for (let i:number=0; i < qtdeRecebida; i++) {
            let indexAleatorio:number;
            do {
                indexAleatorio = Math.floor(Math.random() * tamanhoMaximo)
            } while (listaTemporaria.includes(listaRecebida[indexAleatorio]))

            listaTemporaria.push(listaRecebida[indexAleatorio])
        }
        return listaTemporaria
    }


}

File: 2025-17-04 AVALIACAO 1 (Parte 2) [Java]/Misc.java
import java.util.ArrayList;
import java.util.Random;

public class Misc {
    // Função que recebe uma lista e um valor: retorna um único elemento ou uma lista com elementos aleatórios não-repetidos a partir da lista recebida.
    public static <T> Object retornarElementosAleatorios(T[] listaRecebida, int qtdeRecebida) {
        int tamanhoMaximo = listaRecebida.length;

        // Criando uma instância única de Random para a função inteira
        Random rand = new Random();

        // Caso precise de um só elemento aleatório da lista inserida:
        if (qtdeRecebida == 1) {
            int indexAleatorio = rand.nextInt(tamanhoMaximo);
            return listaRecebida[indexAleatorio];    // Retorna o único elemento aleatório
        }

        // Caso contrário (para qtdeRecebida > 1):
        ArrayList<T> listaTemporaria = new ArrayList<>();
        for (int i = 0; i < qtdeRecebida; i++) {
            int indexAleatorio;
            do {
                indexAleatorio = rand.nextInt(tamanhoMaximo);
            } while (listaTemporaria.contains(listaRecebida[indexAleatorio]));

            listaTemporaria.add(listaRecebida[indexAleatorio]);
        }
        return listaTemporaria;  // Retorna a lista de itens aleatórios
    }
}
